### PR TITLE
[wasm] Unify boot config location

### PR DESCRIPTION
**Pull request by maraf**
- We need both mono-config.json and blazor.boot.json schemas until we can use new SDK in tree
- Rename mono-config to blazor.boot.json to unify location where to load config from
- Drop the need for `withStartupOptions` to use Wasm SDK path in JS

### CODE REVIEW

File: src/mono/wasm/runtime/blazor/BootConfig.ts
Comment (pavelsavara) [inline src/mono/wasm/runtime/blazor/BootConfig.ts:1]: this file should not be here anymore, see loader folder

File: src/mono/wasm/runtime/blazor/_Integration.ts
Comment (pavelsavara) [inline src/mono/wasm/runtime/blazor/_Integration.ts:1]: loader/blazor folder

File: src/mono/wasm/runtime/loader/config.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ import BuildConfiguration from "consts:configuration";
 import type { DotnetModuleInternal, MonoConfigInternal } from "../types/internal";
 import type { DotnetModuleConfig } from "../types";
 import { exportedRuntimeAPI, loaderHelpers, runtimeHelpers } from "./globals";
-import { loadBootConfig } from "./blazor/_Integration";
+import { initializeBootConfig, loadBootConfig } from "./blazor/_Integration";
+import { BootConfigResult } from "./blazor/BootConfig";
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/loader/config.ts:9]: import from wrong file

File: src/mono/wasm/runtime/loader/run.ts [hunk 2]
```diff
@@ -404,7 +404,13 @@ function initializeModules(es6Modules: [RuntimeModuleExportsInternal, NativeModu
 async function createEmscriptenMain(): Promise<RuntimeAPI> {
     if (!module.configSrc && (!module.config || Object.keys(module.config).length === 0 || !module.config.assets)) {
         // if config file location nor assets are provided
-        module.configSrc = "./mono-config.json";
+        if (loaderHelpers.scriptDirectory.indexOf("/_framework") == -1) {
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/loader/run.ts:407]: is this the other way around ? i'm confused about it

Comment (pavelsavara) [reply src/mono/wasm/runtime/loader/run.ts:407]: `loaderHelpers.scriptDirectory` is absolute location of new `dotnet.js` loader.

Comment (maraf) [reply src/mono/wasm/runtime/loader/run.ts:407]: "If scriptDirectory doesn't contain `_framework`, use path with `_framework`"

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**